api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -78,4 +79,69 @@ public PatientQueue getIncompletePatientQueue(Patient patient, Location location
 		return dao.getIncompletePatientQueue(patient, locationTo);
 	}
 	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#getMostRecentQueue(org.openmrs.Patient)
+	 */
+	@Override
+	public PatientQueue getMostRecentQueue(Patient patient) {
+		return dao.getMostRecentQueue(patient);
+	}
+	
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#assignVisitNumber(org.openmrs.module.patientqueueing.model.PatientQueue)
+	 */
+	@Override
+	public PatientQueue assignVisitNumber(PatientQueue patientQueue) {
+		Date today = new Date();
+		List<PatientQueue> patientQueueList = getPatientQueueList(null, OpenmrsUtil.firstSecondOfDay(today),
```

Comment:
To make this clearer, can you wrap it in a method with name like getPatientQueueForToday?

Reply:
@dkayiwa it think it might end up confusing since it assigns. the actual task of it is to assign maybe a modification to cater for today aspect in the method name. but get woudn't be the name to go with